lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -760,7 +760,9 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
             event.defer()
             return
 
-        if plugin_needs_restart or perf_profile_needs_restart:
+        if self.opensearch.is_service_started() and (
```

Comment:
QQ: in general, when to use `is_started` vs `is_service_started`? as we have both

Reply:
We want to know if the JVM is up, not if the service is up. Because `is_started` checks the API call is responsive whereas the `is_service_started` checks the systemd service. We may hit the cluster on a moment where the API is not yet responsive, for whatever reason; but the JVM is already up.